### DISCUSSION TITLE
Options

### DISCUSSION THREAD

**Discussion by kianzarrin** [CitiesSkylinesMods/TMPE#1447] [General]
@aubergine10 I want to know what is your plan for options.
Are you going to store global options in xml file? what about in game options?
Are you going to allow user to change options in main menu to apply to all new games?

**Comment by krzychu124** [comment]
AFAIK, for now it's just refactor to make everything more maintainable and readable but I'm also interested what else is planned in the near future.

**Comment by kianzarrin** [comment]
I always wanted xml options both for in game and for global but its low priority.

**Comment by originalfoo** [comment]
Some of the issues linked in following comment illustrate general trajectory: https://github.com/CitiesSkylinesMods/TMPE/issues/1356#issuecomment-1030504790

For now I'm just moving everything over to consistent format but ultimately I'd like to see options stored as XML, in particular that would make it much easier to do things like global defaults and also 'option presets' (eg. to mimic road policies for certain countries/cities etc).

Next thing I'm looking to do is create a `DropDownOption` UI component to decruft drop-downs in the options.

**Reply by kianzarrin** [reply]
wow you want to fix all of those?
This one also included in that list talks about changing options for new games from main menu: https://github.com/CitiesSkylinesMods/TMPE/issues/562

**Reply by originalfoo** [reply]
Yup, all of them. I need to finish refactor Policies tab first (already done most of it) then a bit of cleanup and we should be in much better position to tackle all those other things.

Regarding #562 having a `Defaults` section in global config would help (at least that's what I'm thinking currently) - a mirror of the mod options fields from `Options.cs`. That will also take us step closer to full transition to xml for `Options.cs` (I think we still need to implement some form of compression though, as TMPE data can grow too big for savegame de/serialization).
